blog/02-02-2023-on-chain-randomness-with-celo-using-subgraphs/on-chain-randomness-with-celo-using-subgraphs.md

```diff
@@ -0,0 +1,1859 @@
+
+    let club = Club.load(event.params.club_.toHexString());
+    winer.club = club!.id;
+    winer.save();
+}
+
+export function handleRewardUpdateToken(event: RewardToken):void {
+    let club = Club.load(event.params.club.toHexString());
+    club!.reward = event.params.reward_;
+    club!.deposit = event.params.deposit_;
+    club!.membersLimit = event.params.membersLimit_;
+    club!.save();
+}
+
+export function handleNFTReward(event: RewardNFT):void {
+    let club = Club.load(event.params.club_.toHexString());
+    club!.reward = event.params.reward_;
+    club!.rewardAddress = event.params.rewardAddress_;
+    club!.save();
+}
+
+export function handleLimitDepoNFTUpdate(event: LimitAndDepoUpdate):void {
+    let club = Club.load(event.params.club_.toHexString());
+    if(event.params.name_ == "membersLimit") {
+        club!.membersLimit = event.params.update_;
+    }else {
+        club!.deposit = event.params.update_;
+    }
+    club!.save();
+}
+```
+
+### Build and Deploy
+
+Once we have completed the code generation process, our next step is to build and deploy the subgraphs. To accomplish this, we can execute the following command in the terminal.
+
+```bash
+# Build
+graph build
+
+
+# Deploy
+graph deploy --product hosted-service <GITHUB_USER>/<SUBGRAPH_NAME>
+```
+
+After deployment, we can access our subgraphs on the dashboard page and test our queries.
+
+![Subgraph Dashboard](images/17.png)
+
+
+## Conclusion
+
+In this tutorial, we have explored two blockchain technologies that can be used to create more transparent and trustworthy applications. Firstly, we learned about the Celo Randomness Contract that enables us to generate safe and fair random numbers on the Celo blockchain. This technology is particularly useful for creating lottery games and other applications that require random numbers.
+
+Next, we delved into Subgraph, a protocol that simplifies creating interactive views and executing queries on the blockchain. With Subgraph, we can design applications that are more interactive and transparent, giving users a clearer and more comprehensive understanding of transactions and activities on the blockchain.
+
+By combining the Celo Randomness Contract and Subgraph, we can develop blockchain applications that are more transparent, fair, and trustworthy, and offer real-time and more interactive data views. By mastering this technology, we can create safer and more sophisticated blockchain applications that users can trust.
+
+
+## Next Step
+
+Setelah anda memahami cara kerja Celo Randomness Contract dan Subgraph, anda dapat mengembangkan aplikasi yang lebih canggih dengan menggunakan kedua teknologi ini. Untuk mempelajari lebih lanjut tentang Celo Randomness Contract dan Subgraph, untuk mempelajari lebih dalam tentang Celo Randomness Contract dan Subgraph, anda dapat mengunjungi [Celo Randomness Contract](https://docs.celo.org/protocol/randomness) dan [Subgraph](https://thegraph.com/docs/en/).
```

Comment:
Can you write this in English, Since its an English tutorial 😅

Reply:
Done, I forgot to translate